### DISCUSSION TITLE
How to use with Vue context ?

### DISCUSSION THREAD

**Discussion by smarlhens** [unjs/unhead#210] [Q&A] [answered]
Hi there 👋🏻 

I hope you are well. Thanks for your job on this library.

I am using this library along `testing-library/vue` and `vitest` to run some tests but since [1.5.0](https://github.com/unjs/unhead/releases/tag/v1.5.0) I'm getting a lot of warnings (`Unhead is missing Vue context, falling back to shared context. This may have unexpected results.`) as described in the release note.
Edit: not only on test but also at runtime I am getting a lot of warnings.

How I use it currently alongside `testing-library/vue`:

<details>

<summary>head.ts</summary>

```typescript
import { createHead } from '@unhead/vue';

export const head = createHead();
```
</details>

<details>

<summary>main.ts</summary>

```typescript
import { head }  from './composables/head';

app.use(head);
```
</details>

<details>

<summary>any-component.vue</summary>

```vue
<script setup lang="ts">
  import { useHead } from '@unhead/vue';

  useHead({
    meta: [{ name: 'description', content: 'This is the best description ever' }],
  });
</script>
```
</details>

<details>

<summary>any-component.test.ts</summary>

```typescript
import type { RenderOptions, RenderResult } from '@testing-library/vue';
import { render } from '@testing-library/vue';
import Component from 'path/to/components/any-component.vue';
import { head }  from 'path/to/composables/head';

describe('any-component.vue', () => {
  let mountOptions: RenderOptions;

  beforeEach(() => {
    mountOptions = {
      global: {
        plugins: [head],
      },
    };
  });

  it('my super test', () => {
    render(Component, mountOptions);

   // [...]
  });
});
```
</details>

Do you have an idea of how to properly inject head and remove these warnings ?
How to provide context to head ?

Thanks for your help 🙏🏻 

Wish you a great day ☀️

**Comment by smarlhens** [comment, marked as answer]
Hi there 👋🏻 

using [1.5.2](https://github.com/unjs/unhead/releases/tag/v1.5.2) `setHeadInjectionHandler` seems to be fixing the issue: 
<details>

<summary>head.ts</summary>

```typescript
import { createHead } from '@unhead/vue';

export const head = createHead();
```
</details>

<details>

<summary>main.ts</summary>

```diff
import { head }  from './composables/head';

- app.use(head);
+ setHeadInjectionHandler(() => head)
```
</details>

<details>

<summary>any-component.vue</summary>

```vue
<script setup lang="ts">
  import { useHead } from '@unhead/vue';

  useHead({
    meta: [{ name: 'description', content: 'This is the best description ever' }],
  });
</script>
```
</details>

<details>

<summary>any-component.test.ts</summary>

```diff
import type { RenderOptions, RenderResult } from '@testing-library/vue';
import { render } from '@testing-library/vue';
import Component from 'path/to/components/any-component.vue';
import { head }  from 'path/to/composables/head';

describe('any-component.vue', () => {
  let mountOptions: RenderOptions;

  beforeEach(() => {
-    mountOptions = {
-      global: {
-        plugins: [head],
-      },
-    };
+    setHeadInjectionHandler(() => head)
+    mountOptions = {};
  });

  it('my super test', () => {
    render(Component, mountOptions);

   // [...]
  });
});
```
</details>

**Comment by harlan-zw** [comment]
Hey @smarlhens. Firstly, thank you for the nice words and detailed discussion :) 

It looks like you spotted my latest release, so well done on fixing it yourself without any docs! Sorry for the delay in getting back to you, wanted to make sure there was a good solution first. 

Wish you a great day as well :sun_with_face:
